### ISSUE TITLE
Add support for EXIF color mamagment for JPEG

### ISSUE THREAD

**Issue by ValZapod** [Beep6581/RawTherapee#6651]
Ultimately https://github.com/Beep6581/RawTherapee/issues/1805 (that was closed wrongly)

With my Nikon D5100 (you need to select Adobe RGB in camera menu). With exiftool:

```
White Point                     : 0.313 0.329
Primary Chromaticities          : 0.64 0.33 0.21 0.71 0.15 0.06
Y Cb Cr Coefficients            : 0.299 0.587 0.114

```
Compare to wikipedia: https://en.wikipedia.org/wiki/Adobe_RGB_color_space#Reference_viewing_conditions

Compared to ICCv4 those values are not chromatically adapted to D50. Those should be used already as is (no, Chrome does not support this yet, but I will look into it, Camera Raw and Photoshop do). There is also the same problem with RAW .NEF images, see  There is another image here: https://github.com/Beep6581/RawTherapee/issues/1805#issuecomment-580203499

![_DSC1521](https://user-images.githubusercontent.com/31514790/190709656-f97c1ce6-860d-4b9b-afc0-dac336b7ad25.JPG)

**Comment by ValZapod** [comment]
Here is the same image with ICC profile, inserted by exiftool, with the same lossless data under, as you can see Chrome sees it now:

![_DSC1521](https://user-images.githubusercontent.com/31514790/209657179-b9cb24d3-f4ad-48b1-80a1-5b7346973ef7.JPG)

**Comment by ValZapod** [comment]
BTW, windows photo viewer also has no problem with it.

**Comment by Entropy512** [comment]
This is probably something that, if any action is taken, really needs to wait until after the exiv2 migration - fixing crap in the existing metadata engine would be wasted time.

There's also the fact that the included metadata is insufficient - it doesn't properly describe the transfer function of the input, which is pretty important.  See some of the discussion in https://discuss.pixls.us/t/doest-it-make-sense-to-process-a-jpeg-camera-file-under-scene-referred-paradigm/34162 - while the thread is discussing darktable, the same concepts (JPEGs need to be properly linearized for processing in a workflow that was designed for raw inputs) apply to RawTherapee.

Because of this, even if RT were to implement the metadata support, it wouldn't work correctly because you'd now have the gamut correct but not the TRC.  So to do it right, you'd still need a custom ICC profile.

As far as AdobeRGB tags being ignored in .NEFs - of course they are.  JPEG colorspace is absolutely meaningless for a raw file.  In the case of RT, since the D5100 is a fully supported camera, all color information comes from the DCP profile in rtdata.  That said, it's a single illuminant DCP, so regenerating it with the more modern dual-illuminant process would be beneficial - we'd need daylight and tungsten reference shots ( see https://rawpedia.rawtherapee.com/How_to_create_DCP_color_profiles ) for that.

**Comment by ValZapod** [comment]
Thankfully I was doing NEF + JPEG mode, cause I thought you would say precisely this. 

[DSC1521.zip](https://github.com/Beep6581/RawTherapee/files/10319838/DSC1521.zip)

![image](https://user-images.githubusercontent.com/31514790/209965027-c78e9153-0093-4e3f-85c0-12c42b748fd2.png)
![image](https://user-images.githubusercontent.com/31514790/209965076-4bbb14c6-3fb3-4516-aa40-5a88706f301f.png)

**Comment by ValZapod** [comment]
>that the included metadata is insufficient - it doesn't properly describe the transfer function of the input, which is pretty important

See latest edit of the issue. Note that Adobe RGB is using 2.20000 pure gamma, no linear part of curve or anything like that, genius Adobe. OETF and EOTF are both 2.20000 pure gamma. That is even more obvious on the other image in https://github.com/Beep6581/RawTherapee/issues/1805#issuecomment-580203499, compare these in Chrome, as you can see the Adobe RGB tagged image has horrible dark details on the left bottom. 

![73444556-bf7b3e00-4393-11ea-803](https://user-images.githubusercontent.com/31514790/209968501-09862a14-178f-4d16-b0eb-c24336ffa120.jpeg)

**Comment by Entropy512** [comment]
> It is there. See latest edit of the issue. Note that Adobe RGB is using 2.20000 pure gamma, no linear part of curve or anything like that, genius Adobe. OETF and EOTF are both 2.20000 pure gamma.

No, it is not.  Please reread the discussion in the link I provided in its entirety.

Especially:
https://discuss.pixls.us/t/doest-it-make-sense-to-process-a-jpeg-camera-file-under-scene-referred-paradigm/34162/8?u=entropy512

and

https://discuss.pixls.us/t/doest-it-make-sense-to-process-a-jpeg-camera-file-under-scene-referred-paradigm/34162/15?u=entropy512

and

https://discuss.pixls.us/t/doest-it-make-sense-to-process-a-jpeg-camera-file-under-scene-referred-paradigm/34162/18?u=entropy512

While these were for a camera saving in sRGB, they all also apply to Adobe RGB - almost no camera on the market actually saves data with a vanilla sRGB or Adobe RGB transfer function.

Using a vanilla transfer function for display works, but it is not suitable for input to RawTherapee's pipeline (which expects scene-linear input, not display-linear).  Proper input to RT's pipeline requires reverse engineering the actual transfer function of the camera and applying that as an ICC profile.

If you really want to use a naive AdobeRGB transfer function, nothing prevents you from using the ICC profile generator to generate an ICC profile describing what you want, and then applying that to your images.

Because of the above, using DCF metadata is going to be a very low priority item for most developers - there are workarounds (applying an ICC profile) AND using it is wrong anyway due to insufficient/wrong metadata regarding transfer function (e.g. the only correct method is to apply an ICC profile that has an actual camera transfer function that has been reverse engineered in some manner, such as the methods I describe in the discussion I linked).  I might take a crack at this *after* the exiv2 switchover, I definitely won't beforehand.  If you want this functionality prior to the exiv2 switchover (e.g. likely 6.0 and not 5.10), you can take a crack at it yourself and submit a PR.

**Comment by ValZapod** [comment]
>Ideally a camera would tag its images with an appropriate ICC profile that described what actually is in the image, but in reality, that is never the case.

See on NX studio on the bottom, slightly to the right. It says it is Nikon sRGB and even has a version number. Some Nikon images are indeed tagged as that icc profile.  See this jpeg: https://github.com/ImageMagick/ImageMagick/issues/5857#issue-1481454079

**Comment by Entropy512** [comment]
That actually just reinforces my point.  If you look at the JPEG, it clearly has an s-curve applied to it.  It doesn't have the "flat" look that something without a tone curve (for example, RT's "neutral" profile applied to a raw image has that "flat" look) has, but yet, this is the embedded TRC:
![image](https://user-images.githubusercontent.com/193570/210253735-078a1d13-1818-48e9-b49d-cadca42ec8ef.png)
I'd need more time to confirm, but that looks like a plain vanilla gamma curve or sRGB TRC - not the S-shape that the camera actually applied to the data.  Per my posts that I linked (again, please read them), this is suitable for display but not suitable for further processing in a scene-linear pipeline.
